modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -205,6 +205,155 @@
 					title: editor.lang.link.title,
 				});
 
+				editor.ui.addRichCombo('LineSpacing', {
+					_applyStyle(className) {
+						var styleConfig = {
+							attributes: {
+								class: className,
+							},
+							element: 'div',
+						};
+
+						var style = new CKEDITOR.style(styleConfig);
+
+						editor.getSelection().lock();
+
+						Object.keys(this._styles).forEach((spacingKey) => {
+							var item = this._styles[spacingKey];
+
+							if (
+								this._checkActive({
+									elementPath: editor.elementPath(),
+									styleConfig: item.style,
+								})
+							) {
+								editor.removeStyle(
+									new CKEDITOR.style(item.style)
+								);
+							}
+						});
+
+						editor.applyStyle(style);
+
+						editor.getSelection().unlock();
+					},
+
+					_checkActive({elementPath, styleConfig}) {
+						var active = true;
+
+						if (elementPath && elementPath.lastElement) {
+							styleConfig.attributes.class
+								.split(' ')
+								.forEach((className) => {
+									active =
+										active &&
+										elementPath.lastElement.hasClass(
+											className
+										);
+								});
+						}
+						else {
+							active = false;
+						}
+
+						return active;
+					},
+
+					_styles: {
+						'1.0x': {
+							style: {attributes: {class: ''}, element: 'div'},
+						},
+						'1.5x': {
+							style: {
+								attributes: {class: 'mt-1 mb-1'},
+								element: 'div',
+							},
+						},
+						'2.0x': {
+							style: {
+								attributes: {class: 'mt-2 mb-2'},
+								element: 'div',
+							},
+						},
+						'3.0x': {
+							style: {
+								attributes: {class: 'mt-3 mb-3'},
+								element: 'div',
+							},
+						},
+						'4.0x': {
+							style: {
+								attributes: {class: 'mt-4 mb-4'},
+								element: 'div',
+							},
+						},
+						'5.0x': {
+							style: {
+								attributes: {class: 'mt-5 mb-5'},
+								element: 'div',
+							},
+						},
+					},
+
+					init() {
+						var instance = this;
+
+						Object.keys(this._styles).forEach((spacingKey) => {
+							instance.add(spacingKey, spacingKey);
+						});
+					},
+
+					label: '1.0x',
+
+					onClick(key) {
+						this._applyStyle(
+							this._styles[key].style.attributes.class
+						);
+
+						var richComboRef = editor.ui.get('LineSpacing');
+
```

Comment:
Why the `Ref` in the name? 
This ain't React :smile: just saying

Reply:
> This ain't React 😄 just saying

~CKEditor~ React Editor™️ 


![](https://media.giphy.com/media/jrfM0X9WADtCriz1Hb/giphy.gif)

Reply:
This is weird. I'm not using a addRichCombo anymore. I need to squash commits for a better visualization :)

Same as https://github.com/liferay-frontend/liferay-portal/pull/1150#discussion_r660872509